jicoco-config/src/test/kotlin/org/jitsi/config/TypesafeConfigSourceTest.kt

```diff
@@ -51,8 +51,23 @@ class TypesafeConfigSourceTest : ShouldSpec() {
                 }
             }
             context("Double") {
-                withConfig("double=42.5") {
-                    getValue<Double>("double") shouldBe 42.5
+                mapOf("42" to 42.0, "42.5" to 42.5, "42.5e-2" to 42.5e-2,
+                      "-42" to -42.0, "-42.5" to -42.5, "-42.5e-2" to -42.5e-2,
+                      "42.5%" to 42.5e-2, "42.5 %" to 42.5e-2, "42 %" to 42e-2,
+                      "-42.5%" to -42.5e-2, "-42.5 %" to -42.5e-2, "-42 %" to -42e-2).forEach { (k, v) ->
```

Comment:
This is making me wonder if we want a separate `percentage` type or something, should a negative percent be valid?

Reply:
No, that's just another way to encode a number (we want users to be able to use `0.05` as well as `5%`). We have `Duration`, not separate types for minutes and seconds, etc.

Reply:
But how would `-5%` be treated?

Reply:
`-0.05`. Is there any other reasonable way?

Reply:
I think we should have made that a failure case.

Reply:
If this is for `%` only (allow `-0.05`, but not `-5%`), it would be inconsistent. If it's for numbers in general, it might break code. We could have a `PositiveNumber` type, or something like that, but that seems unrelated.